### ISSUE TITLE
Infinite recursion when using `passthrough: true`

### ISSUE THREAD

**Issue by bengotow** [electron/electron-compile#149]
If you specify `"text/less": { passthrough: true }`, electron-compile runs the passthrough compiler, which returns "the non-final MIME type text/less". This causes electron-compile to get stuck infinitely running the passthrough compiler trying to get a final MIME type.

It seems there are two sources of "finalness" in CompilerHost: `finalForms` and  `CompilerHost.shouldPassthrough`. Neither of these return true for `text/less` when you request that it be passed through.

**Comment by bengotow** [comment]
I worked around this on my own branch by exiting if the file went through compilation without changing mimetypes: https://github.com/bengotow/electron-compile/commit/ef8ed4554a3162381a72ddf5c7a909b8163a0862.

I'm not sure if this is a good general solution - really depends what the typical use cases are for recursive compilation (minfication, anything else?)

**Comment by crdrost** [comment]
Yeah, I don't know if I can bump this to the devs' attention in any way but this is a serious bug and the documentation advocating `{passthrough: true}` should be immediately deleted if this bug is not going to be fixed any time soon.

**Comment by MarshallOfSound** [comment]
If @bengotow can submit his changes in a PR we'll probably take them 😄

**Comment by crdrost** [comment]
Since @bengotow basically disabled recursive compilation wholesale I wrote something quick that is a little more granular: it only disables recursive compilation when a very large number of passes occur or when one of those passes produces identical source to a prior pass.
